Code cleanup

himdel opened the pull request:
* Use `os.getenv`, not `os.environ.get`; drop default if None (consistency only)
* Cleanup more dead code/props
* Change `f'{val}'` to `val` or `str(val)`
* Separate `get_install_type` from `config` collector
* No `price_per_node` in renewal guidance
* `build_report`: use more readable variables; use `date.today` instead of `datetime.now`; move extractor to after `extra_params` is fully set
* `validate_ship_path`: don't mutate global const, remove unused gather code path

himdel committed:
Remove dead code

himdel committed:
Separate get_install_type from config collector

himdel committed:
Change f'{val}' to val; still stringify price_per_node for report header

himdel committed:
No price_per_node in renewal guidance

himdel committed:
build_report: make nicer, use date.today instead of datetime.now, and move extractor to after extra_params is fully set

himdel committed:
validate_ship_path: don't mutate global var, replace unused gather code path

himdel committed:
unrelated, cosmetic

MilanPospisil commented on metrics_utility/management/commands/build_report.py:
```diff
@@ -99,14 +99,14 @@ def handle(self, *args, **options):
         extra_params['month_since'] = month
         extra_params['month_until'] = next_month
-        extra_params['deduplicator'] = os.getenv('METRICS_UTILITY_DEDUPLICATOR', None) or None
+        extra_params['deduplicator'] = os.getenv('METRICS_UTILITY_DEDUPLICATOR') or None
```
Is it even necessary? This should return None by default if variable not found.

Habbit from javascript? :-)

himdel replied:
If not found, yeah, but `METRICS_UTILITY_DEDUPLICATOR= command` will find an empty string, so this just turns `""` into `None`

MilanPospisil replied:
This is evil...

MilanPospisil commented:
Needs rebase.

MilanPospisil commented on metrics_utility/management/commands/build_report.py:
```diff
@@ -109,24 +110,26 @@ def handle(self, *args, **options):
-            extra_params['since_date'] = opt_since.date()
-            extra_params['until_date'] = opt_until.date() if opt_until else now.date()
+            since_date = opt_since.date()
+            until_date = opt_until.date() if opt_until else datetime.date.today()
```
This is practicaly removing datetime granularity and setting date granularity.

Are you sure we will not need it in service? I can imagine we will be building reports for hours.

MilanPospisil commented on metrics_utility/management/validation.py:
```diff
@@ -276,16 +276,13 @@ def validate_ship_path(errors, ship_target, method):
-    ship_path = os.getenv('METRICS_UTILITY_SHIP_PATH', no_path)
-    dir_paths = VALID_SHIP_TARGET_BUILD
-    if 's3' in dir_paths:
-        dir_paths.remove('s3')
```
Oh, this mutates original VALID_SHIP_TARGET_BUILD, nice catch!

himdel committed:
Unify on os.getenv, with default=None omitted

os.environ.get -> os.getenv

    ast-grep -p 'os.environ.get($$$A)' -r 'os.getenv($$$A)'

os.getenv(key, default=None) - drop default when None

    ast-grep -p 'os.getenv($A, None)' -r 'os.getenv($A)'

+make fix

MilanPospisil reviewed (approved):
LGTM :+1:
